nanoFramework.Json/TimeExtensions.cs

```diff
@@ -109,15 +137,47 @@ public static DateTime FromIso8601(string date)
 			string second = (parts.Length > 5) ? parts[5] : "0";
 			string ms = (parts.Length > 6) ? parts[6] : "0";
 
+			// If the input string is non-numeric, Convert.ToInt32() below will throw an exception from the Native code
+			// These exceptions cause annoying delays & output when running in the nanoFramework debugger
+			// Bail out if the input string is non-numeric
+			if (!IsNumeric(year))
+			{
+				return DateTime.MaxValue;
+			} 
+			else if (!IsNumeric(month))
+			{
+				return DateTime.MaxValue;
+			} 
+			else if (!IsNumeric(day))
+			{
+				return DateTime.MaxValue;
+			} 
+			else if (!IsNumeric(hour))
+			{
+				return DateTime.MaxValue;
+			} 
+			else if (!IsNumeric(minute))
+			{
+				return DateTime.MaxValue;
+			} 
+			else if (!IsNumeric(second))
+			{
+				return DateTime.MaxValue;
+			} 
+			else if (!IsNumeric(ms))
+			{
+				return DateTime.MaxValue;
+			}
+
 			// sanity check for bad milliseconds format
 			int milliseconds = Convert.ToInt32(ms);
 
-			if(milliseconds > 999)
+            if (milliseconds > 999)
```

Comment:
Still some alignment problems in the code. really use Visual Studio 2019, the community version is free and use the magic combination of `ctrl + k + d`

Reply:
Done.  I've reset my VS options to the default for C# but still not sure if ctrl + k + d is working correctly.